### PR TITLE
Lesson03

### PR DESCRIPTION
**Pull request by ilyatoka**
Take a look please )

### CODE REVIEW

File: lesson03/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+// Concatenate .js to signle file and generate sourcemaps
+gulp.task('js', function () {
+    return gulp.src('js/**/*.js', {base: '.'})
+        .pipe(gulpif(!argv.prod, sourcemaps.init()))
+        .pipe(concat('js/main.js'))
+        .pipe(gulpif(!argv.prod, sourcemaps.write()))
+        .pipe(gulp.dest(destDir));
+});
+
+gulp.task('compress-js', function (callback) {
```
Comment (kobylin) [inline lesson03/gulpfile.js]: This task doesn't called anywhere. Could you make `uglify()` to be part of `js` task, which runs with `--prod` flag? Like in `css` task

File: lesson03/gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
+var gulp = require('gulp'),
+    bower = require('gulp-bower'),
+    gulpif = require('gulp-if'),
+    concat = require('gulp-concat'),
+    less = require('gulp-less'),
+    argv = require('yargs').argv,
+    debug = require( 'gulp-debug' ),
+    clean = require( 'gulp-clean' ),
+    livereload = require('gulp-livereload'),
```
Comment (kobylin) [inline lesson03/gulpfile.js]: You forgot include `gulp-livereload` into package.json :)

File: lesson03/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+gulp.task('htmlhint', function () {
+    return gulp.src(['**/*.html', '!' + destDir + '/**/*.*', '!node_modules/**/*.*', '!libs/**/*.*'])
+        .pipe(htmlhint('.htmlhintrc'))
+        .pipe(htmlhint.reporter())
+        .pipe(htmlhint.failReporter({suppress: true}));
+});
+
+// Check and fix codestyle for .js
+gulp.task('jscs', function () {
+    return gulp.src(destDir + 'js/**/*.js')
```
Comment (kobylin) [inline lesson03/gulpfile.js]: You need to check source js files for codestyle, but not compiled

File: lesson03/gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        .pipe(jscs())
+        .pipe(jscs.reporter())
+        .pipe(jscs.reporter('fail'));
+});
+
+// Check and log errors in .js
+gulp.task('jshint', function () {
```
Comment (kobylin) [inline lesson03/gulpfile.js]: You need to check source js files for codestyle, but not compiled

File: lesson03/gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+// Ensures the `images` task is complete before reloading browsers
+gulp.task('images-watch', ['images'], function (done) {
+    browserSync.reload();
+    done();
+});
+
+// Copy all .html file to destDir. Run with flag --prod to minify .html files
+gulp.task('html', function () {
+    return gulp.src(['**/*.html', '!' + destDir + '/**/*.*', '!node_modules/**/*.*', '!libs/**/*.*'], {base: '.'})
+        .pipe(gulpif(argv.prod, htmlmin({collapseWhitespace: true})))
```
Comment (kobylin) [inline lesson03/gulpfile.js]: Could you also add flag for removing comments from html?

File: lesson03/gulpfile.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+});
+
+// Compile .less files and generate sourcemaps. Run with flag --prod to minify output .css
+gulp.task('css', function () {
+    return gulp.src('styles/**/*.less')
+        .pipe(gulpif(!argv.prod, sourcemaps.init()))
+        .pipe(concat('styles/styles.css'))
+        .pipe(less())
+        .pipe(gulpif(argv.prod, minifyCss()))
+        .pipe(prefix("last 3 version"))
```
Comment (kobylin) [inline lesson03/gulpfile.js]: I'd suggest you to add support of `>5%, ie9-11` browsers. 
And use some `css3` styles(gradient for background, border-radius etc.) to see real benefit of prefixer in `bin/styles/styles.css`

File: lesson03/gulpfile.js (hunk 2 above)
Comment (kobylin) [inline lesson03/gulpfile.js]: Or remove it, cause I see you are not using it

File: lesson03/gulpfile.js (hunk 1 above)
Comment (ilyatoka) [inline lesson03/gulpfile.js]: Yeah, you're right! I removed `compress-js` task completely and moved `uglify()` inside `js` task as you mentioned.

File: lesson03/gulpfile.js (hunk 2 above)
Comment (ilyatoka) [inline lesson03/gulpfile.js]: It lost forever now )

File: lesson03/gulpfile.js (hunk 3 above)
Comment (ilyatoka) [inline lesson03/gulpfile.js]: Oops! Got it and fixed!

File: lesson03/gulpfile.js (hunk 4 above)
Comment (ilyatoka) [inline lesson03/gulpfile.js]: Oops! Got it and fixed!

File: lesson03/gulpfile.js (hunk 5 above)
Comment (ilyatoka) [inline lesson03/gulpfile.js]: Flag added.

File: lesson03/gulpfile.js (hunk 6 above)
Comment (ilyatoka) [inline lesson03/gulpfile.js]: Ok, task updated. And yes, autoprefixer works awesome, like always ))

**Comment by kobylin** [comment]
Great work! You can merge it :+1:

**Pull request merged**